File: SQLiteRepository/SQLiteQuery/UnitOfWork.cs
using System;
using System.Collections.Generic;
using System.Data.SQLite;
using System.Linq;
using SQLiteRepository.SQLiteQuery;

namespace SQLiteRepository.SQLiteQuery
{
    public class UnitOfWork
    {
        private readonly List<ISqliteQuery> _queries = new List<ISqliteQuery>();
        public List<ISqliteQuery> Queries
        {
            get { return _queries; }
        }

        public List<IQueryResult> Execute()
        {
            var cumresult = new List<IQueryResult>();

            var conn = GetConnection(_queries);
            conn.Open();
            using (var tr = conn.BeginTransaction())
            {
                try
                {
                    foreach (var sqliteQuery in _queries)
                    {
                        var result = sqliteQuery.Execute(conn,tr);
                       cumresult.Add(result);
                    }
                    tr.Commit();
                }
                catch (Exception ex)
                {
                    tr.Rollback();
                    throw;
                }
                finally
                {
                    conn.Close();
                }
            }


            return cumresult;
        }


        private static SQLiteConnection GetConnection(IEnumerable<ISqliteQuery> queries)
        {
            var firstQuery = queries.FirstOrDefault();

            if (firstQuery == null) return null;
            var conn = CommonSqlite.CreateConnection(firstQuery.Path);
            return conn;
        }
    }
}

File: SQLiteAdapters/UnitOfWork.vb
Imports System
Imports System.Collections.Generic
Imports System.Data.SQLite
Imports System.Linq


Public Class UnitOfWork
    Public Property Queries As List(Of ISqliteQuery)

    Public Function Execute() As List(Of IQueryResult)
        Dim cumresult = New List(Of IQueryResult)()

        Dim conn = GetConnection(Queries)
        conn.Open()
        Using tr = conn.BeginTransaction()
            Try
                For Each sqliteQuery In Queries
                    Dim result = sqliteQuery.Execute(conn, tr)
                    cumresult.Add(result)
                Next
                tr.Commit()
            Catch ex As Exception
                tr.Rollback()
                Throw
            Finally
                conn.Close()
            End Try
        End Using


        Return cumresult
    End Function


    Private Shared Function GetConnection(queries As IEnumerable(Of ISqliteQuery)) As SQLiteConnection
        Dim firstQuery = queries.FirstOrDefault()

        If firstQuery Is Nothing Then
            Return Nothing
        End If
        Dim conn = CommonSqlite.CreateConnection(firstQuery.Path)
        Return conn
    End Function
End Class